modules/common/src/main/scala/surge/streams/EventSink.scala

```diff
@@ -20,11 +22,29 @@ trait EventHandler[Event] {
 }
 
 trait EventSink[Event] extends EventHandler[Event] {
+  val tracer: Tracer
+  def sinkName: String = this.getClass.getSimpleName
   def parallelism: Int = 8
   def handleEvent(key: String, event: Event, headers: Map[String, Array[Byte]]): Future[Any]
   def partitionBy(key: String, event: Event, headers: Map[String, Array[Byte]]): String
 
+  // execute the handleEvent function with tracing
+  private def execHandleEvent(key: String, event: Event, headers: Map[String, Array[Byte]]): Future[Any] = {
+    import surge.internal.tracing.TracingHelper._
+    val operationName = s"$sinkName:${event.getClass.getSimpleName}"
+    val span = tracer.spanBuilder(operationName).setNoParent().startSpan()
```

Comment:
Is there a decent way to detect if this span follows from another one? I think it'd be more a follows_from than a parent child relationship in the source->sink case

This handler logic is ultimately hooked into an `EventSource` which provides the message reading/deserialization pieces of a stream processor. Eventually I think we'd want the span to trace all the way from there - for example, start when a message is read from Kafka, send to the sink here and continue the trace, additionally carry through the sink side for certain sinks we own (like the command engine sink since we've instrumented all the way through that)

Reply:
Yes, working on it